### ISSUE TITLE
js files always installed

### ISSUE THREAD

**Issue by 0-wiz-0** [transmission/transmission#4864]
### What is the issue?

I've just updated the package for transmission in pkgsrc to 4.0.0.
We have separate packages for the daemon and the gtk and qt frontends. We want each installed file to belong only to one package.

I didn't find an option to stop CMake from installing the following files:
```
share/transmission/public_html/images/favicon.ico
share/transmission/public_html/images/favicon.png
share/transmission/public_html/images/webclip-icon.png
share/transmission/public_html/index.html
share/transmission/public_html/transmission-app.js
share/transmission/public_html/transmission-app.js.LICENSE.txt
```
I had assumed that `-DENABLE_WEB=OFF` would stop the installation, but it didn't.
Did I overlook an option, or is this a bug?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.0

**Comment by ckerr** [comment]
Background info: `ENABLE_WEB` is a misleading name; it's a leftover from earlier versions but in 4.0.0 it's more a flag of "do or do not rebuild the generated files from scratch" since build boxes may not have access to new versions of Node.js and/or may not have network access. See #4760.

So I don't think it's a _bug_ except that the name is misleading -- but we didn't consider your use case of multiple packages.

What did you have in mind; a common package that all three would depend on? Because -daemon, -gtk , and -qt can all serve these files.

**Comment by ckerr** [comment]
The shortest path to disabling that install is in the top-level CMakeLists.txt:

```cmake
if(ENABLE_DAEMON OR ENABLE_GTK OR ENABLE_QT)
    tr_install_web(${CMAKE_INSTALL_DATAROOTDIR}/${TR_NAME})
endif()
```

I'm open to suggestions / PRs on how to make this better to address the packaging use case.

**Comment by t-8ch** [comment]
FYI in ArchLinux the web stuff is part of the package transmission-cli on which the GUIs carry optional dependencies.

**Comment by t-8ch** [comment]
I'll probably have time in the coming days.

But I'll have to check why the current behavior is an issue for pkgsrc.
It seems weird to me that this is an issue.

@0-wiz-0 do you have a link to your buildscripts?

**Comment by 0-wiz-0** [comment]
@t-8ch Thanks!
The problem is quite easy to understand even without the build scripts - see https://github.com/transmission/transmission/issues/4864#issuecomment-1428185948 above - the same files are installed if either the daemon, the gtk frontend, or the qt frontend are built. These are separate packages in pkgsrc whose lists of installed files shouldn't overlap so you can install some or all of them in parallel - a file must only belong to one package in pkgsrc.

If you want to look at the build scripts anyway, they are here:
https://github.com/NetBSD/pkgsrc/tree/trunk/net/transmission
https://github.com/NetBSD/pkgsrc/tree/trunk/net/transmission-gtk
https://github.com/NetBSD/pkgsrc/tree/trunk/net/transmission-qt
but the mirror is slightly outdated and still shows the 3.x build scripts - it should catch up soon.

**Comment by t-8ch** [comment]
Ah, I naively only thought about split packages from a single build. My bad, it makes sense now.

**Comment by ckerr** [comment]
Removing `4.0.1` milestone. This would  be nice-to-have to make packagers' lives easier but it's not a blocker if the PR doesn't arrive in time
